### ISSUE TITLE
RF06: Unnecessary quoted identifier fails for reserved quote keyword

### ISSUE THREAD

**Issue by yichilin** [sqlfluff/sqlfluff#5407]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Due to some reasons, I need to use SQL reserved keyword `date` as column name so I quote as `"date"`. 
When I run `sqlfluff linting` on the file, it fails with `Unnecessary quoted identifier "date".` 

Example SQL
```sql
SELECT
    CAST(start_date AS date) AS "date"
...
```

### Expected Behaviour

According to the document,
```text
When prefer_quoted_identifiers = False (default behaviour), the quotes are unnecessary, except for reserved keywords and special characters in identifiers.
```

I expect it should not fail since date is a reserved keyword and I already quoted it. 

### Observed Behaviour

It fails with `Unnecessary quoted identifier "date".` 

### How to reproduce

Just create simple query with the following statement and then run ` sqlfluff lint`
```sql
SELECT '2023-11-15' AS "date"
```

### Dialect

redshift

### Version

2.1.2

### Configuration

[sqlfluff]
templater = dbt
dialect = redshift
exclude_rules = RF02
max_line_length = 100
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = false
indented_using_on = true
template_blocks_indent = false

[sqlfluff:templater]
unwrap_wrapped_queries = true

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

[sqlfluff:layout:type:comma]
line_position = leading

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by kokorin** [comment]
```
[sqlfluff:rules:references.quoting]
prefer_quoted_keywords = true
```
Allows using quoted keywords in  SQLFluff 3.5

**Comment by peterbud** [comment]
Also AWS Redshift is a **lowercase** dialect, so the linter deems `"date"` and `date` equivalent.
